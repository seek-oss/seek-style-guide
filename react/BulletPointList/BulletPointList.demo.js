import React from 'react';
import { BulletPointList, Text } from 'seek-style-guide/react';
import * as sketch from './BulletPointList.sketch';

export default {
  route: '/bullet-point-list',
  title: 'Bullet Point List',
  category: 'Typography',
  component: BulletPointList,
  sketch,
  initialProps: {
    children: [
      /* eslint-disable react/jsx-key */
      <Text>Bullet point 1</Text>,
      <Text>Bullet point 2</Text>,
      <Text>Bullet point 3</Text>
    ]
  },
  options: []
};
