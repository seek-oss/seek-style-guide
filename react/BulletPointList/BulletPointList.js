import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import BulletPoint from '../BulletPoint/BulletPoint';

import styles from './BulletPointList.less';

const renderChild = child => {
  if (!React.isValidElement(child)) {
    return undefined;
  }
  const key = child.props.key || child.props.index;
  return <BulletPoint key={key}>{child}</BulletPoint>;
};

export default function BulletPointList({ className, children }) {
  return (
    <ul className={classNames(styles.root, className)}>
      {React.Children.map(children, renderChild)}
    </ul>
  );
}

BulletPointList.displayName = 'BulletPointList';

BulletPointList.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
