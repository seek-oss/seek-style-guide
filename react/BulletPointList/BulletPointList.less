@import (reference) '~seek-style-guide/theme';

.root {
  margin-bottom: 3 * @row-height;
}
