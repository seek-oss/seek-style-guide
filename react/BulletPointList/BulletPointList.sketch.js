import React from 'react';
import { BulletPointList, BulletPoint, Text } from 'seek-style-guide/react';
export const symbols = {
  'BulletPoint/List': (
    <BulletPointList>
      <Text>Bullet point 1</Text>
      <Text>Bullet point 2</Text>
      <Text>Bullet point 3</Text>
    </BulletPointList>
  ),
  ButtonPoint: <BulletPoint>Bullet point</BulletPoint>
};
