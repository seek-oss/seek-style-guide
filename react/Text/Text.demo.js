import Text from './Text';
import * as sketch from './Text.sketch';

export default {
  route: '/text',
  title: 'Text',
  category: 'Typography',
  component: Text,
  sketch,
  initialProps: {
    children: 'Living Style Guide',
    hero: true
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Strong',
          transformProps: props => ({
            ...props,
            strong: true
          })
        },
        {
          label: 'Regular',
          transformProps: props => ({
            ...props,
            regular: true
          })
        },
        {
          label: 'Light',
          transformProps: props => ({
            ...props,
            light: true
          })
        },
        {
          label: 'Highlight',
          transformProps: props => ({
            ...props,
            highlight: true
          })
        },
        {
          label: 'Disable baseline',
          transformProps: props => ({
            ...props,
            baseline: false
          })
        }
      ]
    },
    {
      label: 'Tone',
      type: 'radio',
      states: [
        {
          label: 'Neutral tone',
          transformProps: props => props
        },
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            positive: true
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            critical: true
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            secondary: true
          })
        },
        {
          label: 'Info',
          transformProps: props => ({
            ...props,
            info: true
          })
        }
      ]
    },
    {
      label: 'Size',
      type: 'radio',
      states: [
        {
          label: 'Hero',
          transformProps: props => props
        },
        {
          label: 'Headline',
          transformProps: ({ hero, ...props }) => ({
            ...props,
            headline: true
          })
        },
        {
          label: 'Heading',
          transformProps: ({ hero, ...props }) => ({
            ...props,
            heading: true
          })
        },
        {
          label: 'Subheading',
          transformProps: ({ hero, ...props }) => ({
            ...props,
            subheading: true
          })
        },
        {
          label: 'Large',
          transformProps: ({ hero, ...props }) => ({
            ...props,
            large: true
          })
        },
        {
          label: 'Standard',
          transformProps: ({ hero, ...props }) => props
        },
        {
          label: 'Small',
          transformProps: ({ hero, ...props }) => ({
            ...props,
            small: true
          })
        }
      ]
    }
  ]
};
