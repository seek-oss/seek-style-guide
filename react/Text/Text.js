import styles from './Text.less';
import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';

import stylesPositive from '../Positive/Positive.less';
import stylesCritical from '../Critical/Critical.less';
import stylesSecondary from '../Secondary/Secondary.less';
import stylesHighlight from '../Highlight/Highlight.less';
import stylesInfo from '../Info/Info.less';
import stylesStrong from '../Strong/Strong.less';
import stylesRegular from '../Regular/Regular.less';

import withTextProps, { SizePropTypes } from '../private/withTextProps';

const Text = ({
  component,
  bullet,
  children,
  className,
  size,
  raw,
  positive,
  critical,
  secondary,
  highlight,
  info,
  strong,
  regular,
  light,
  baseline,
  ...restProps
}) => {
  const Component = component || (bullet ? 'li' : 'span');
  return (
    <Component
      {...restProps}
      className={classnames({
        [styles.root]: true,
        [className]: className,
        [styles[size]]: size,
        [styles.raw]: raw,
        [styles.bullet]: bullet,
        [styles.baseline]: baseline
      })}
    >
      <span
        className={classnames({
          [stylesPositive.root]: positive,
          [stylesCritical.root]: critical,
          [stylesSecondary.root]: secondary,
          [stylesHighlight.root]: highlight,
          [stylesHighlight.secondary]: highlight && secondary,
          [stylesInfo.root]: info,
          [stylesStrong.root]: strong,
          [stylesRegular.root]: regular,
          [styles.light]: light
        })}
      >
        {children}
      </span>
    </Component>
  );
};

Text.displayName = 'Text';

Text.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
    PropTypes.func
  ]),
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  bullet: PropTypes.bool,
  ...SizePropTypes,
  raw: PropTypes.bool,
  positive: PropTypes.bool,
  critical: PropTypes.bool,
  secondary: PropTypes.bool,
  highlight: PropTypes.bool,
  info: PropTypes.bool,
  strong: PropTypes.bool,
  regular: PropTypes.bool,
  light: PropTypes.bool,
  baseline: PropTypes.bool
};

Text.defaultProps = {
  baseline: true,
  bullet: false,
  size: 'standard'
};

export default withTextProps(Text);
