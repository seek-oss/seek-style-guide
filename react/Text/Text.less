@import (reference) "~seek-style-guide/theme";

.root {
  padding-bottom: @row-height * 2;
  display: block;

  .root {
    position: relative;
    outline: 2px solid red;
    &::after {
      content: "ERROR: Multiple <Text> components nested";
      background: red;
      font-weight: @sk-bold;
      color: @sk-white;
      position: absolute;
      top: 0;
      right: 0;
      padding: 0 @gutter-width;
    }
  }
}

.standard {
  &:not(.baseline) {
    .standardTextResponsive(@baseline: false);
  }
  &.baseline {
    .standardTextResponsive(@baseline: true);
  }
  padding-bottom: @row-height;
}

.small {
  &:not(.baseline) {
    .smallTextResponsive(@baseline: false);
  }
  &.baseline {
    .smallTextResponsive(@baseline: true);
  }
  padding-bottom: @row-height;
}

.large {
  &:not(.baseline) {
    .largeTextResponsive(@baseline: false);
  }
  &.baseline {
    .largeTextResponsive(@baseline: true);
  }
  padding-bottom: @row-height;
}

.subheading {
  &:not(.baseline) {
    .subheadingTextResponsive(@baseline: false);
  }
  &.baseline {
    .subheadingTextResponsive(@baseline: true);
  }
}

.headline {
  &:not(.baseline) {
    .headlineTextResponsive(@baseline: false);
  }
  &.baseline {
    .headlineTextResponsive(@baseline: true);
  }
}

.heading {
  &:not(.baseline) {
    .headingTextResponsive(@baseline: false);
  }
  &.baseline {
    .headingTextResponsive(@baseline: true);
  }
}

.hero {
  &:not(.baseline) {
    .heroTextResponsive(@baseline: false);
  }
  &.baseline {
    .heroTextResponsive(@baseline: true);
  }
}

.light {
  font-weight: @sk-light;
}

.raw {
  padding-bottom: 0;
}

.bullet {
  display: list-item;
  list-style-type: disc;
  margin-left: 1.5 * @gutter-width;
}
