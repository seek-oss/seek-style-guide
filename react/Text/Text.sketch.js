import React from 'react';
import mapKeys from 'lodash/mapKeys';
import lessToJs from 'less-vars-to-js';
import type from '!!raw-loader!seek-style-guide/theme/type/type.less';
import Text from './Text';
import TextLink from '../TextLink/TextLink';

const sizes = [
  'hero',
  'headline',
  'heading',
  'subheading',
  'large',
  'standard',
  'small'
];
const typeVars = lessToJs(type);
const baseFontSize = parseInt(typeVars['@base-font-size'], 10);
const symbolNames = sizes.reduce((obj, size, i) => {
  const [firstChar, ...restChars] = size;
  const name = `${firstChar.toUpperCase()}${restChars.join('')}`;

  const desktopSize =
    parseFloat(typeVars[`@${size}-type-scale`]) * baseFontSize;
  const mobileSize =
    parseFloat(typeVars[`@${size}-type-scale-mobile`]) * baseFontSize;

  const symbolName = `${i +
    1}. ${name} â€” Desktop: ${desktopSize}px, Mobile: ${mobileSize}px`;

  return { ...obj, [size]: symbolName };
}, {});

export const text = {
  [`${symbolNames.hero}`]: <Text hero>Hero text</Text>,
  [`${symbolNames.headline}`]: <Text headline>Headline text</Text>,
  [`${symbolNames.heading}`]: <Text heading>Heading text</Text>,
  [`${symbolNames.subheading}`]: <Text subheading>Subheading text</Text>,
  [`${symbolNames.large}`]: <Text large>Large text</Text>,
  [`${symbolNames.standard}/1. Default`]: <Text>Standard text</Text>,
  [`${symbolNames.standard}/2. Secondary`]: (
    <Text secondary>Standard critical text</Text>
  ),
  [`${symbolNames.standard}/3. Strong`]: (
    <Text strong>Standard strong text</Text>
  ),
  [`${symbolNames.standard}/4. Positive`]: (
    <Text positive>Standard positive text</Text>
  ),
  [`${symbolNames.standard}/5. Critical`]: (
    <Text critical>Standard critical text</Text>
  ),
  [`${symbolNames.standard}/6. Link`]: (
    <Text>
      <TextLink href="#">Standard link text</TextLink>
    </Text>
  ),
  [`${symbolNames.small}`]: <Text small>Small text</Text>
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(text, (value, key) => `Text/${key}`);
