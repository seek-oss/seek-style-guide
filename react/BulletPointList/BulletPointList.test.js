import React from 'react';
import { shallow } from 'enzyme';

import Text from '../Text/Text';

import BulletPointList from './BulletPointList';

describe('BulletPointList', () => {
  it('renders with standard text', () => {
    const wrapper = shallow(
      <BulletPointList>
        <Text>Foo</Text>
        <Text>Bar</Text>
      </BulletPointList>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
