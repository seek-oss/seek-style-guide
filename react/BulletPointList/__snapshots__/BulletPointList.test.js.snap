// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPointList renders with standard text 1`] = `
<ul
  className="root"
>
  <BulletPoint
    key=".0"
  >
    <Text>
      Foo
    </Text>
  </BulletPoint>
  <BulletPoint
    key=".1"
  >
    <Text>
      Bar
    </Text>
  </BulletPoint>
</ul>
`;
